.github/scripts/create_alerts.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import os
+import re
+from collections import defaultdict
+from difflib import SequenceMatcher
+from typing import Any, Dict, List, Set, Tuple
+
+import requests
+from setuptools import distutils  # type: ignore[import]
+
+ALL_SKIPPED_THRESHOLD = 100
+SIMILARITY_THRESHOLD = 0.75
+FAILURE_CHAIN_THRESHOLD = 2
+MAX_CONCURRENT_ALERTS = 1
+FAILED_JOB_PATTERN = (
+    r"^- \[(.*)\]\(.*\) failed consecutively starting with commit \[.*\]\(.*\)$"
+)
+
+PENDING = "pending"
+NEUTRAL = "neutral"
+SKIPPED = "skipped"
+SUCCESS = "success"
+FAILURE = "failure"
+CANCELED = "canceled"
+
+ISSUES_WITH_LABEL_QUERY = """
+query ($owner: String!, $name: String!, $labels: [String!]) {
+  repository(owner: $owner, name: $name, followRenames: false) {
+    issues(last: 10, labels: $labels, states: [OPEN]) {
+      nodes {
+        id
+        title
+        closed
+        number
+        body
+        createdAt
+        comments(first: 100) {
+          nodes {
+            bodyText
+            databaseId
+          }
+        }
+      }
+    }
+  }
+}
+"""
+
+NUM_ISSUES_QUERY = """
+query ($query: String!) {
+  search(type: ISSUE, query: $query) {
+    issueCount
+  }
+}
+"""
+
+REPO_OWNER = "pytorch"
+PYTORCH_REPO_NAME = "pytorch"
+TEST_INFRA_REPO_NAME = "test-infra"
```

Comment:
I'm in favor of creating the issue in PyTorch instead of TEST_INFRA_REPO_NAME, but I'm not sure how an issue on PyTorch is synced to internal and picked up by the you-know-what rule.

Reply:
I think I can delete that bit, we can decide later of where to stick the alerts. I was thinking for now just put everything in test-infra.